components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,408 @@
+		this.expanded = false;
+		this.headingLevel = defaultHeading;
+		this.headingStyle = defaultHeading;
+		this.padding = 'default';
+		this.type = 'default';
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._intersectionObserver) this._intersectionObserver.disconnect();
+	}
+
+	firstUpdated() {
+		if (typeof(IntersectionObserver) === 'function') {
+			this._intersectionObserver = new IntersectionObserver(entries => {
+				if (!this.expanded) return;
+
+				const entry = entries[0];
+				const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+				if (entry.isIntersecting) {
+					element.classList.remove('scrolled');
+				} else {
+					element.classList.add('scrolled');
+				}
+			});
+			this._intersectionObserver.observe(this.shadowRoot.querySelector('.d2l-collapsible-panel-top-sentinel'));
+		}
+	}
+
+	render() {
+		const expandCollapseLabel = this.expandCollapseLabel || this.panelTitle;
+
+		return html`
+			<button
+				aria-expanded="${this.expanded}"
+				class="d2l-offscreen"
+				type="button"
+				@click="${this._toggleExpand}"
+				@focus="${this._onFocus}"
+				@blur="${this._onBlur}"
+			>${expandCollapseLabel}</button>
+			<div class="d2l-collapsible-panel" @click="${this._handlePanelClick}">
+				<div class="d2l-collapsible-panel-top-sentinel"></div>
+				${this._renderHeader()}
+				<d2l-expand-collapse-content ?expanded="${this.expanded}">
+					<div class="d2l-collapsible-panel-content">
+						<slot></slot>
+					</div>
+				</d2l-expand-collapse-content>
+				<d2l-expand-collapse-content ?expanded="${!this.expanded}">
+					<div class="d2l-collapsible-panel-summary">
+						<slot name="summary" @slotchange="${this._handleSummarySlotChange}"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+			</div>
+		`;
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('expanded')) {
+			if (!this.expanded) {
+				const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+				element.classList.remove('scrolled');
+			}
+
+			const event = `d2l-collapsible-panel-${this.expanded ? 'expand' : 'collapse' }`;
+
+			this.dispatchEvent(new CustomEvent(
+				event, { bubbles: false, composed: false }
+			));
+		}
+	}
+
+	_handleActionsClick(e) {
+		const actions = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-actions');
+		if (e.target !== actions) {
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderClick(e) {
+		if (this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderSecondaryClick(e) {
+		const header = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-secondary');
+		if (e.target !== header) {
+			e.stopPropagation();
+		}
+	}
+
+	_handlePanelClick(e) {
+		if (!this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleSummarySlotChange() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('has-summary');
```

Comment:
Ditto here I think in that this should be a state-based `hasSummary` property that causes the class to be added/removed in `render()`.